### DISCUSSION TITLE
Use standard ports 80, 443?

### DISCUSSION THREAD

**Discussion by jeffshead** [rustdesk/rustdesk#984] [Q&A] [answered]
Maybe I missed it but I haven't found any info or instructions on making RD use standard ports. Most properly configured LAN's in a business environment limit end user traffic in/out of their gateway firewall to ports 80, 443 and maybe a couple others.

Has anyone installed RD server and set it up to use standard ports so that you can remotely connect and support end users (behind firewalls blocking all but port 80 and 443) over the Internet?

**Comment by ghost** [comment]
I second this. I can use Teamviewer at my office to connect to my home server, but not RustDesk for this very reason.

**Comment by deajan** [comment]
Configuring HBBS & HBBR to listen on standard ports is easy.
But where to configure rustdesk client to use them ?
Any news on this ?

Ports 2111x aren't generally open outbound on any network. Thus using standard ports 80 & 443 is a reasonable claim.

**Comment by pedjas** [comment]
I support this. Only ports 80 and 443 are safe to use. Others may be (and often are) blocked by routers.
The key to success of remote access platform is its ability to work without requirement for special settings on routers.

**Comment by phyzjqkl** [comment]
no one can answer this?

**Comment by deajan** [comment]
Looks like the [documentation](https://rustdesk.com/docs/en/self-host/install/#step-3--set-hbbshbbr-address-on-client-side) now show how to change hbbr and hbbs client side ports

As for the hbbs and hbbr server modules, just NAT them via your local firewall.

**Reply by shensmobile** [reply]
This is only possible when you self-host right?  You can't connect to the public servers using that technique since you can't actually specify them in the ID/Relay server fields, unless there's a list of their IP/Web addresses out there?

**Comment by phyzjqkl** [comment]
yes. but what is the procedure?

获取Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Reply by shensmobile** [reply]
To be honest, I'm not sure.  I'm tinkering with it now but I'm not convinced that this will allow us to force the client to listen/communicate through ports 443 and 80.

**Comment by phyzjqkl** [comment]
I think if i am the owner of host, i can let user know i am using private server. i wonder for portable host (run rustdesk without install), it can talk to name server with 80 ports? I hv a job site computer which need to be host but only 80 port open

**Comment by deajan** [comment]
I am too trying to use standard ports right now.
I can successfully connect to my selfhosted rustdesk server using hbbs port 21116.
I've made a NAT src port 21116 dst port 21116 to my hbbs server.
I've also made a NAT src port 80 dst port 21116 to my hbbs server.
Both NATs are for TCP and UDP connections.

Rustdesk logs show that the hbbs connection work on both ports 21116 or 80, but the desktop app writes `Not ready. Please check your connection` when using port 80. I've also tried port 443, same results.

I'm quite puzzled actually. I think rustdesk client calculates other ports by doing addiing or substracting integers to the current hbbs port number, hence rendering it not usable for anything else.
Anyone got rustdesk to work with standard ports ?

I've also changed hbbs port with `--port 443` directive in hbbs executable, and made NAT port 443:443. No results.

My firewall sees incomming TCP connections on port 443 but never any UDP connections. Of course they're allowed.
Using standard port 21116 in rustdesk client will trigger both TCP and UDP connections to port 21116.

Created an issue at https://github.com/rustdesk/rustdesk/issues/4260

**Comment by deajan** [comment]
My issue has been closed with a `wontfix` badge, without any further explanation.
There has to be an obvious problem, but I don't really get it.

**Reply by rustdesk** [reply]
I am sorry to make you unhappy, to avoid confusion, we make this decision, https://github.com/rustdesk/rustdesk/issues/4331, this discussion will be the first one.

**Reply by deajan** [reply]
> I am sorry to make you unhappy, to avoid confusion, we make this decision, #4331, this discussion will be the first one.

Okay, makes sense.
I'll allow myself to copy the issue text here.

**Comment by deajan** [comment]
Since rustdesk works with specific ports 21115-21119, a great number of public wifi networks as well as some corporate environment will not allow rustdesk connections.

Most networks always allow ports 80, 110, 143 and 443 but filter anything else.
It would be great if rustdesk could make usage of those default ports, 80 and 443 being the best, for both self hosted or public hbbs/hbbr servers.

As ideal scenario, rustdesk would try to connect to port 21116, and fallback to 443 then to 80 if not successful.
Teamviewer for example uses this exact connection schema, with a proprietary port and then some fallback ports, see https://community.teamviewer.com/English/kb/articles/4139-ports-utilises-par-teamviewer

Anybody that ever used a filtered network like a wifi hotspot in a café in Europe for example will enjoy this feature ;-D
It doesn't seem that rustdesk can be configured to use those ports properly as of today.

**Comment by Kofl** [comment]
Absolutely, it would increase the usage of Rustdesk tremendous with fallback to 443/80
All commercial products like Teamviewer, AnyDesk, WiseConnect use that technique as nowadays outgoing any ports is rare.

**Comment by IAMBUDE** [comment]
This missing feature sadly is a deal breaker for us.
I'm still looking for a decent self-hosted remote support solution and rustdesk was fairing great so far but many of our customers block outgoing connections except for http or https.

**Comment by kragil** [comment]
This should be the default.

**Comment by My1** [comment]
the fun question is also is it using Protocols a firewall can understand (even if it cant see the actual contents because encryption) because not sure if firewalls check for the traffic being HTTP and HTTPS (or at the very least TLS in general) on 80&443

**Reply by ben-kuhn** [reply]
Good firewalls inspect at layer 7 (application) as well as layer 3 (TCP or UDP port) so just changing the port isn't always going to help.  The firewall will see non-HTTP(S) traffic on 80 or 443 and drop it.

**Comment by Logic-Bits** [comment]
we also experienced some problems not using the "default" open Ports... then always have to switch to TeamViewer.

**Comment by djordans** [comment]
Hello,
Ports 443 and 80 work but UDP 443 does not work on all internet connections
I tested with Server ID on 443 and Relay on 80, it works but you have to open ports 443 TCP & UDP on the firewall

**Reply by deajan** [reply]
Considering HTTP/2 uses TCP and UDP, that's a fair configuration that should be used widely.

**Reply by Kofl** [reply]
Yes, but note, that Enterprise Networks block http/2 (UDP) as most security appliances are still not yet able to scan the traffic.

**Comment by CyrosX** [comment]
Any news on this? Is it possible with the new HTTP proxy?

**Comment by deajan** [comment]
Indeed, not being able to use ports 80/443 is the only reason that keeps me from using Rustdesk...
I am puzzled as of why the devs never made any statement about this.

**Comment by mscheltienne** [comment]
Same need here, `rustdesk` can't be used on my office network since the ports are blocked. I have to fall back to `teamviewer`..

**Comment by C4T-B** [comment]
I just set up a Server and everything for our Company and just realized i can't change these weird Ports. seriously boggles my mind, now i can bin the entire Project

**Comment by danielchristianschroeter** [comment]
Using standard ports 80 and 443 for RustDesk makes a lot of sense for several reasons:

1. Firewall Compatibility: Most business environments and public networks restrict outbound traffic to a few standard ports, primarily 80 (HTTP) and 443 (HTTPS). By using these ports, RustDesk can bypass these restrictions, ensuring seamless connectivity even in highly secured networks.
2. Ease of Configuration: Configuring firewalls to allow traffic through non-standard ports can be complex and time-consuming. Using standard ports simplifies this process, making it easier for IT administrators to deploy and manage RustDesk.
3. Reliability: Standard ports are less likely to be blocked by routers and firewalls. This increases the reliability of remote connections, as users won’t face connectivity issues due to port restrictions.
4. Industry Practice: Many commercial remote access solutions like TeamViewer and AnyDesk use fallback mechanisms to connect through standard ports if their primary ports are blocked. Adopting a similar approach would make RustDesk more competitive and user-friendly.
5. Security: Ports 80 and 443 are generally considered safe and are monitored closely for security threats. Using these ports can leverage existing security measures in place for web traffic, adding an extra layer of protection.

Implementing support for these ports would significantly enhance RustDesk’s usability and appeal, especially in environments with strict network policies.

**Comment by rustdesk** [comment]


**Reply by 1plus-1** [reply]
Sometimes using RustDesk through proxy is not possible. For my case all the companies I've been working have the popular proxy tools blocked. I understand maybe using UDP can improve the performance. But not everyone concerned the performance a lot. Sometimes the basic connectivity is more important. At least user can have the configurable option to sacrifice performance to get basic access.

**Reply by Kofl** [reply]
and additionally unfortunately its only supported with the pro version

**Reply by OtterKring** [reply]
But why use something additional instead of implementing it as a default?

**Comment by mscheltienne** [comment]
Thanks for the links. Seeing as other commercial remote access solutions, TeamViewer, AnyDesk, work "out of the box", wouldn't a fallback mechanism to get RustDesk to work out of the box too be a must to increase adoption?

**Comment by devNull2and1** [comment]
@rustdesk we are pro users now since 2 years, also accepted new prices for your licenses and reordered...in hope that you will redesign the port structure. As already dicussed here, proxy is no solution as it would be blocked to. At least normal Ports should be possible but no!
Even if i configure:

rdi.domain.com:443 -> pub ip 1.2.3.4:443 -> hdi:21116
rdr.domain.com:443 -> pub ip 1.2.3.5:443 -> hdr:21117

the client connects 443-1 to port 442. Why? Why? At least it should be possible to configure that he should not calculate 443-1. Then it would be posstible to say 1.2.3.4:80 -> hdi:21115, or whatever.

Best Approach would be one port that handles http or https, like an gateway. just a third container hdg that pipes everything over one port.
This is the last step of perfection for RustDekst to beat TeamViewer and AnyDesk. Is this step impossible?

**Comment by rustdesk** [comment, marked as answer]
The web socket only solution (for Pro users) is added this week. You do not need to shout any more. :(

We prepared a video tutrial for you. https://x.com/rustdesk/status/1923379412617064489

You can also follow this guide: https://github.com/rustdesk/rustdesk/wiki/FAQ#websocket

BTW, you can also self-host web client (Pro server 1.5.9), `https://rustdesk.yourcompany.com/web` works for you.

**Reply by devNull2and1** [reply]
> The web socket only solution (for Pro users) is added this week. You do not need to shout any more. :(
> 
> We prepared a video tutrial for you. https://x.com/rustdesk/status/1923379412617064489
> 
> You can also follow this guide: https://github.com/rustdesk/rustdesk/wiki/FAQ#websocket
> 
> BTW, you can also self-host web client (Pro server 1.5.9), `https://rustdesk.yourcompany.com/web` works for you.

Shout? Please don't get me wrong. I've held back on this topic for two years, posting a maximum of two times, and the whole time I just hoped something would come along. You're making me very happy with this information. I don't fucking want anything else than Rustdesk.

**Reply by rustdesk** [reply]
OSS users will shout then. :(

Why isn't this feature available to free users as well? Will any one fork this repo and add this feature?

Hope one day Trump gives me some money (a little bit he earned in the stock market last month), then I can work for the world for free.

**Reply by devNull2and1** [reply]
I understand, but not anything can be free. I use RustDeskt Pro in Office and Also RustDesk for private things without pro. 
In private environments, these ports aren't blocked, are they? Your extra work is rewarded by the Pro version and its license fees. Thats it!

**Reply by Bouby308** [reply]
How can I enable WebSocket only version for iOS? Custom client is not supported on iOS.

**Reply by rustdesk** [reply]
> How can I enable WebSocket only version for iOS? Custom client is not supported on iOS.

Will be in 1.4.1. I might try to push a 1.4.0-1 for iOS.

**Reply by silversparker** [reply]
> > How can I enable WebSocket only version for iOS? Custom client is not supported on iOS.
> 
> Will be in 1.4.1. I might try to push a 1.4.0-1 for iOS.

I know that a lot of work has gotten into the iOS app lately, is there a rough eta for a hotfix or the next version? we're willing to switch over to pro as soon as the websocket support is done
